BSPConversionLib/Source/EntityConverter.cs

```diff
@@ -137,6 +143,82 @@ private HashSet<string> GetGiveTargets()
 			return targets;
 		}
 
+		private void ConvertFuncDoor(Entity entity)
+		{
+			SetMoveDir(entity);
+
+			if (float.TryParse(entity["health"], out var health))
+				entity.ClassName = "func_button"; // Health is obsolete on func_door, maybe fix in engine and update this
+		}
+
+		private void ConvertFuncButton(Entity entity)
+		{
+			SetMoveDir(entity);
+
+			SetButtonFlags(entity);
+
+			if (entity["wait"] == "-1") // A value of -1 in quake is instantly reset position, in source it is don't reset position.
+				entity["wait"] = "0.001"; // exactly 0 also behaves as don't reset in source, so the delay is as short as possible without being 0.
+
+			var targets = GetTargetEntities(entity);
+
+			foreach (var target in targets)
+			{
+				switch (target.ClassName)
+				{
+					case "func_door":
+						OpenDoorOnPressed(entity, target);
+						break;
+				}
+			}
+		}
+
+		private static void OpenDoorOnPressed(Entity button, Entity door)
+		{
+			var connection = new Entity.EntityConnection()
+			{
+				name = "OnPressed",
+				target = door["targetname"],
+				action = "Open",
+				param = null,
+				delay = 0,
+				fireOnce = -1
+			};
+			button.connections.Add(connection);
+		}
+
+		private static void SetButtonFlags(Entity button)
+		{
+			float.TryParse(button["speed"], out var speed);
+			if (!float.TryParse(button["speed"], out _))
+				return;
+
+			if ((speed == -1 || speed >= 9999) && (button["wait"] == "-1")) // TODO: Add customization setting for the upper bounds potentially?
+				button["spawnflags"] = "1"; // Don't move flag
+
+			if (!float.TryParse(button["health"], out var health) || button["health"] == "0")
+				button["spawnflags"] = "256"; // Press on touch
+		}
+
+		private static void SetMoveDir(Entity entity)
+		{
+			float.TryParse(entity["angle"], out var angle);
+			
+			if (!float.TryParse(entity["speed"], out _))
+				return;
+
+			else if (angle == -1) // UP
```

Comment:
Change `else if` to `if`. This condition has nothing to do with the previous TryParse